packages/devtools_app/lib/src/screens/deep_link_validation/deep_links_controller.dart

```diff
@@ -207,7 +207,7 @@ class DeepLinksController extends DisposableController
         },
         associatedPath: [
           ...previousRecord?.associatedPath ?? [],
-          if (linkData.path != null) linkData.path!,
+          if (!(linkData.path?.isExcluded ?? true)) linkData.path!.path,
```

Comment:
sorry for the churn here. In my original comment, I didn't realize that the value was negated, so that being said I think it is more readable the way you had it originally:
`if (linkData.path != null && !linkData.path!.isExcluded)`